### DISCUSSION TITLE
Inconsistent number of columns

### DISCUSSION THREAD

**Discussion by ASPCKenneth** [JoshClose/CsvHelper#2271] [General]
Hi

I am using the reader to parse a csv file  and do not want the reading to stop if an inconsistent amount of columns was detected.
So I handle the ReadingExceptionOccurred and return false, so far so good.

But the next row, which is valid, again raises this exception.

This looks like a bug to me?
In ReadAsync (and Read) methods you can find following, and imho the columncount should not always be set, just compare the parser.count to the amount of columns of the header/first row - or at least only set it when it did not encounter a difference?

```
	if (detectColumnCountChanges && hasMoreRecords)
	{
		if (columnCount > 0 && columnCount != parser.Count)
		{
			var csvException = new BadDataException(string.Empty, parser.RawRecord, context, "An inconsistent number of columns has been detected.");

			var args = new ReadingExceptionOccurredArgs(csvException);
			if (readingExceptionOccurred?.Invoke(args) ?? true)
			{
				throw csvException;
			}
		}

		columnCount = parser.Count;
	}


```

**Comment by ASPCKenneth** [comment]
I managed to work around this in the exception handler and compare there the current parser.count with the header count (both are available in the ReadingExceptionOccurredArgs.Exception.Context

but still.

**Comment by JoshClose** [comment]
`DetectColumnCountChanges` is `false` by default. Are you setting it to `true`?
